string str
---
float32 dist
